backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Context.kt

```diff
@@ -277,6 +277,12 @@ internal class Context(config: KonanConfig) : KonanBackendContext(config) {
     var serializedIr: SerializedIrModule? = null
     var dataFlowGraph: ByteArray? = null
 
+    val moduleId by lazy {
+        irModule!!.name.asString().localHash.value.toInt()
```

Comment:
Using an address of a private LLVM global as a TLS map key is kinda more flexible:
1.  It would work when compiling only a part of a module (e.g. proper incremental compilation).
2.  It would make TLS code generation more self-contained (e.g. independent from IR).